<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BSplineEditor</class>
 <widget class="QDialog" name="BSplineEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>B-Spline Editor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QGraphicsView" name="graphicsView">
      <property name="minimumSize">
       <size>
        <width>400</width>
        <height>400</height>
       </size>
      </property>
      <property name="renderHints">
       <set>QPainter::Antialiasing|QPainter::SmoothPixmapTransform</set>
      </property>
      <property name="dragMode">
       <enum>QGraphicsView::ScrollHandDrag</enum>
      </property>
      <property name="transformationAnchor">
       <enum>QGraphicsView::AnchorUnderMouse</enum>
      </property>
      <property name="resizeAnchor">
       <enum>QGraphicsView::AnchorUnderMouse</enum>
      </property>
      <property name="viewportUpdateMode">
       <enum>QGraphicsView::SmartViewportUpdate</enum>
      </property>
     </widget>
     <widget class="QWidget" name="controlPanel" native="true">
      <property name="minimumSize">
       <size>
        <width>200</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="anchorsGroupBox">
         <property name="title">
          <string>Anchor Points</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QPushButton" name="addAnchorButton">
              <property name="text">
               <string>Add Point</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="removeAnchorButton">
              <property name="text">
               <string>Remove Point</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QFormLayout" name="formLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="xLabel">
              <property name="text">
               <string>X:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QDoubleSpinBox" name="xSpinBox">
              <property name="minimum">
               <double>-1000.000000000000000</double>
              </property>
              <property name="maximum">
               <double>1000.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="yLabel">
              <property name="text">
               <string>Y:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QDoubleSpinBox" name="ySpinBox">
              <property name="minimum">
               <double>-1000.000000000000000</double>
              </property>
              <property name="maximum">
               <double>1000.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QListWidget" name="pointsListWidget"/>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="splineGroupBox">
         <property name="title">
          <string>B-Spline Properties</string>
         </property>
         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="nSegmentsLabel">
            <property name="text">
             <string>N Segments:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="nSegmentsSpinBox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>10</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="viewControlsGroupBox">
         <property name="title">
          <string>View Controls</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QPushButton" name="zoomInButton">
              <property name="text">
               <string>Zoom In</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="zoomOutButton">
              <property name="text">
               <string>Zoom Out</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPushButton" name="resetViewButton">
            <property name="text">
             <string>Reset View</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QCheckBox" name="autoApplyCheckBox">
         <property name="text">
          <string>Auto Apply</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QPushButton" name="applyButton">
           <property name="text">
            <string>Apply</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="okButton">
           <property name="text">
            <string>OK</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="cancelButton">
           <property name="text">
            <string>Cancel</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
